### DISCUSSION TITLE
Missing step in 'Cleaning up a Cluster' in the docs

### DISCUSSION THREAD

**Discussion by peterroth** [rook/rook#12856] [Q&A]
Hello everybody,
I just had the opportunity to get to know Rook Ceph (operator v1.12.2, ceph v17.2.6) and installed it on an OKD cluster, then removed, following the [official documentation](https://rook.io/docs/rook/v1.12/Getting-Started/ceph-teardown/). I removed the services that were using Ceph provided PVs, then followed with the step where the cephblockpool should be removed:
`oc delete -n rook-ceph cephblockpool myblockpool`
At this point, the prompt stuck for about 2 hours, nothing happened. I checked and finalizer is _cephblockpool.ceph.rook.io_. When I removed the finalizer, the "myblockpool" was removed immediately, but later when I installed Ceph again it found leftover things from the previous installation, as expected.
At the next try, when I was about to remove the "myblockpool", I left the command run for 3,5 hours, but had the same situation as before, it was stuck, waiting for something. I didn't want to run into the previous failure, so I searched on the net how to remove the cephblockpool so I can go further with the steps and clean up Ceph completely and successfully. I found the [Delete a Pool](https://access.redhat.com/documentation/en-us/red_hat_ceph_storage/1.2.3/html/storage_strategies/delete-a-pool) guide on RedHat's site. I created the rook-ceph-tools pod, entered into it and ran the command mentioned in that guide, got the `pool 'myblockpool' removed` about immediately, then ran the `oc delete -n rook-ceph cephblockpool myblockpool` against my OKD cluster and 5 seconds later, the resource was deleted.
My question is: how can I acknowledge the risk and mark the cephblockpool to be deletable? Is there any other way to do it? If not, can the official documentation enhanced with this necessary step?

**Comment by subhamkrai** [comment]
@peterroth please look at this dependencies chart https://github.com/rook/rook/blob/master/design/ceph/resource-dependencies.md#inventory-of-resources-that-could-have-dependency-relationships you first need to remove the children resource or resource having ownerreferences on another resource(parent resource)

**Reply by peterroth** [reply]
Thank you for the answer, @subhamkrai,
The first step in the Cleanup documentation is to remove the pods claiming the PVs. IT is logical, and that's what I also did.
Additionally, if there were still resources that use the parent resource (PVCs and PVs in this case), I shouldn't be able to acknowledge the deletion from the Toolbox pod, removing the cephblockpool from ceph itself.

**Reply by subhamkrai** [reply]
@peterroth currently, where are you stuck? which resources are not getting deleted?

**Reply by peterroth** [reply]
At the 3rd command in the [Delete the Block and File artifacts](https://rook.io/docs/rook/v1.12/Getting-Started/ceph-teardown/#delete-the-block-and-file-artifacts) part: `oc delete -n rook-ceph cephblockpool myblockpool`
There **were** no PVCs, no pods that **were** using the above-mentioned blockpool, still it couldn't be deleted until I entered the rook-ceph-tools pod and initiated the `ceph osd pool delete myblockpol myblockpool --yes-i-really-really-mean-it` command, acknowledging the loss of data residing on the blockpool.
